framework/include/systems/SystemBase.h

```diff
@@ -443,23 +443,39 @@ class SystemBase : public libMesh::ParallelObject, public ConsoleStreamInterface
   /**
    * Gets a reference to a variable of with specified name
    *
+   * This excludes and cannot return finite volume variables.
+   *
    * @param tid Thread id
    * @param var_name variable name
    * @return reference the variable (class)
    */
   template <typename T>
   MooseVariableFE<T> & getFieldVariable(THREAD_ID tid, const std::string & var_name);
 
+  /**
+   * Returns a field variable pointer - this includes finite volume variables.
+   */
+  template <typename T>
+  MooseVariableField<T> & getActualFieldVariable(THREAD_ID tid, const std::string & var_name);
```

Comment:
:rofl: ugh this is terrible. As someone who has gone through the hell of fixing all the apps, I understand this decision. I feel like with releases we shouldn't have to make this kind of design decision anymore. We should be able to make `getFieldVariable` do the right thing (e.g. change its meaning from its current meaning to its post-FV meaning) and then application developers would be responsible for updating to the new meaning of the API

Reply:
hahaha